workflows/postprocessing.nf

```diff
@@ -135,19 +136,27 @@ workflow POSTPROCESSING {
             updated_meta["id"] = updated_meta.familyId
             [updated_meta, vcf.flatten(), tbi.flatten()]}
     
-    filtered_one = filtered.filter{it[0].sampleSize == 1}.map{meta,vcf,tbi -> [meta,[vcf[0],tbi[0]]]}
+    filtered_one = filtered.filter{it[0].sampleSize == 1}//.map{meta,vcf,tbi -> [meta,[vcf[0],tbi[0]]]}
     filtered_mult = filtered.filter{it[0].sampleSize > 1}
     //Combine per-sample gVCF files into a multi-sample gVCF file
     
-    DB = COMBINEGVCFS(filtered_mult, pathReferenceGenomeFasta,pathReferenceGenomeFai,pathReferenceDict,pathIntervalFile)
+    ch_combined_gvcf = COMBINEGVCFS(filtered_mult, pathReferenceGenomeFasta,pathReferenceGenomeFai,pathReferenceDict,pathIntervalFile)
                     .combined_gvcf.join(COMBINEGVCFS.out.tbi)
-                    .map{meta, gvcf,tbi -> [meta,[gvcf, tbi]]}
                     .concat(filtered_one)
 
+    geno_input_files = ch_combined_gvcf.map{meta,vcf,tbi -> [meta, vcf, tbi, [], []]}
+    geno_input_fasta = ch_combined_gvcf.map{meta,vcf,tbi -> [meta,pathReferenceGenomeFasta]}
+    geno_input_fai = ch_combined_gvcf.map{meta,vcf,tbi -> [meta,pathReferenceGenomeFai]}
+    geno_input_dict = ch_combined_gvcf.map{meta,vcf,tbi -> [meta,pathReferenceDict]}
+    geno_input_dbsnp = ch_combined_gvcf.map{meta,vcf,tbi -> [meta,[]]}
+    geno_input_dbsnpidx = ch_combined_gvcf.map{meta,vcf,tbi -> [meta,[]]}
     //Perform joint genotyping on one or more samples
-    vcf = genotypeGVCF(DB, referenceGenome)
+    genotypegvcf_output = GATK4_GENOTYPEGVCFS(geno_input_files,geno_input_fasta,geno_input_fai,geno_input_dict,geno_input_dbsnp,geno_input_dbsnpidx).vcf
```

Comment:
Je ne suis pas certaine qu'il faut faire l'appel avec des channels séparés de cette manière.  Je ne suis pas sûre que nextflow va combiner les channels  correctement. Je vais vérifier en local de mon côté par précaution. Si nous ne faisons pas la bonne chose, ça pourrait entraîner des problèmes de corruption. Ça va me prendre un peu de temps. Je te redonne des nouvelles.

Reply:
J'ai regardé la documentation de Nextflow et il semble que l'utilisation de plusieurs channels de type queue dans un même process n'est pas recommandée. Voici une note que j'ai trouvée dans la documentation :

"In general, multiple input channels should be used to process combinations of different inputs, using the each qualifier or value channels. Having multiple queue channels as inputs is equivalent to using the merge operator, which is not recommended as it may lead to non-deterministic process inputs."

Reply:
J'essaie de comprendre comment on est supposés appeler le process et l'intention derrière les variables meta2, meta3, ... meta6 en extra ... Elles ne sont pas utilisés dans le process du tout. Elles ont été ajoutées récemment, je crois il y a une semaine ou deux.

Reply:
Ce code marche pour moi en local:
```
GATK4_GENOTYPEGVCFS(
        geno_input_files,
        Channel.value([[:], pathReferenceGenomeFasta]),
        Channel.value([[:], pathReferenceGenomeFai]),
        Channel.value([[:], pathReferenceDict]),
        Channel.value([[:], []]), //leaving empty as we don't use dbsnp
        Channel.value([[:], []])  //leaving empty as we don't use dbsnp
    )
```

Reply:
Ça fonctionne aussi sans l'appel à Channel.value(). Je crois qu'il fait la conversion automatiquement:

```
    genotypegvcf_output = GATK4_GENOTYPEGVCFS(
        geno_input_files,
        [[:], pathReferenceGenomeFasta],
        [[:], pathReferenceGenomeFai],
        [[:], pathReferenceDict],
        [[:], []], //leaving empty as we don't use dbsnp
        [[:], []]  //leaving empty as we don't use dbsnp
    ).vcf
```